PushSharp.Apple/AppleHttpPushChannel.cs

```diff
@@ -0,0 +1,200 @@
+                }
+
+                if (response.IsSuccessStatusCode)
+                {
+                    if (callback != null)
+                        callback(this, new SendNotificationResult(notification));
+                }
+                else
+                {
+                    var responseBody = response.Content.ReadAsStringAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+                    Log.Error("Error during APNS Send with channel {0}: {1} -> Code {2} - {3}",
+                                        _channelInstanceId,
+                                        appleNotification.Identifier,
+                                        response.StatusCode,
+                                        string.IsNullOrEmpty(responseBody) ? "No response body" : responseBody);
+
+                    if (response.StatusCode == HttpStatusCode.Gone && callback != null)
+                    {
+                        var result = new SendNotificationResult(notification, false, new Exception("Device token no longer available on APNs."))
+                        {
+                            IsSubscriptionExpired = true,
+                            OldSubscriptionId = appleNotification.DeviceToken,
+                            SubscriptionExpiryUtc = DateTime.UtcNow
+                        };
+                        callback(this, result);
+                        return;
+                    }
+
+                    if (callback != null)
+                    {
+                        callback(this, new SendNotificationResult(notification, false,
+                                    new Exception("Error during APNS Send.", new Exception(string.Format("Error during APNS Send with channel {0}: {1} -> Code {2} - {3}",
+                                        _channelInstanceId,
+                                        appleNotification.Identifier,
+                                        response.StatusCode,
+                                        string.IsNullOrEmpty(responseBody) ? "No response body" : responseBody)))));
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error("Exception during APNS Send with channel {2}: {0} -> {1}",
+                    appleNotification.Identifier,
+                    ex,
+                    _channelInstanceId);
+
+                if (callback != null)
+                {
+                    // will requeue the notification
+                    callback(this, new SendNotificationResult(notification, true, ex));
+                }
+            }
+        }
+
+        private string GetJwtToken()
+        {
+            if (!string.IsNullOrEmpty(_currentJWT))
+            {
+                // checks expiration (50m) - APNs expects the token to have a TTL between 20m and 60m
+                var tokenExpiration = DateTime.UtcNow - _JWTCreationDate.Value;
+                if (tokenExpiration.TotalMinutes < 50)
```

Comment:
By storing ```_JWTCreationDate``` we have to calculating the total minutes since creation for every notification. 
If we calculate the 'JWTExpiryTime' instead (Now() + 50 minutes when creating a token) we can avoid the date calculations on every notification send, and just do a simple compare (Now() > Expiry?)